Pylint should not emit `logging-not-lazy` when using explicit string concatenation for the message

### Bug description

```python
# pylint: disable=missing-docstring
import logging

logging.error('first' + 'second')
```

Since there is no difference in byte code whether the string is explicitly concatenated with `+` or not, the logging isn't really not lazy.

Extra context: the _Black_ formatter will by default wrap long strings in parens with extra indentation:

```pyformat
logging.info("this is a super super super super super super super super super super super super super long message: %s", content)
```

is reformatted to:

```python
logging.info(
    (
        "this is a super super super super super super super super super super super"
        " super super long message: %s"
    ),
    content,
)
```

Not all people like the extra indentation level, so they could workaround this by using explicit str concatenations:

```python
logging.info(
    "this is a super super super super super super super super super super"
    + " super super super long message: %s",
    content
)
```

Above code won't be wrapped. See example: https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AGJAJZdAD2IimZxl1N_Wl9XMO7It8_a9_zDM7vdFRRO_5rVL_TbaRMZFsldltUpcVRrU3ciQwYZYh1fEUKKhGKajA7blNBzUd1tD9xf20knxX9nUEnMw4r5um_wmnJAbkpayNa6vKQJW31lZlMZ-h8FDJRG0yvSqO-tn_wDhAqHt1M950Ia2LlS8KBt25Oi9a35Qd-DQeQ5aXTpAAAAAJCojbjuFauOAAGyAYoDAAAVp5UpscRn-wIAAAAABFla

But this triggers the `logging-not-lazy` error where it really shouldn't.

### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
************* Module flag3app
a.py:4:0: W1201: Use lazy % formatting in logging functions (logging-not-lazy)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```


### Expected behavior

It should not raise `logging-not-lazy`

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.11.2 (main, Mar  3 2023, 16:02:46) [GCC 12.2.0]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8410.false_positive
+++ b/doc/whatsnew/fragments/8410.false_positive
@@ -1,0 +1,3 @@
+`logging-not-lazy` is not longer emitted for explicitly concatenated string arguments.
+
+Closes #8410

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -293,7 +293,7 @@ class LoggingChecker(checkers.BaseChecker):
     def _is_operand_literal_str(operand: InferenceResult | None) -> bool:
         """Return True if the operand in argument is a literal string."""
         return isinstance(operand, nodes.Const) and operand.name == "str"
-    
+
     @staticmethod
     def _is_node_explicit_str_concatenation(node: nodes.NodeNG) -> bool:
         """Return True if the node represents an explicitly concatenated string."""

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -246,7 +246,7 @@ class LoggingChecker(checkers.BaseChecker):
         if isinstance(format_arg, nodes.BinOp):
             binop = format_arg
             emit = binop.op == "%"
-            if binop.op == "+":
+            if binop.op == "+" and not self._is_node_explicit_str_concatenation(binop):
                 total_number_of_strings = sum(
                     1
                     for operand in (binop.left, binop.right)
@@ -293,6 +293,19 @@ class LoggingChecker(checkers.BaseChecker):
     def _is_operand_literal_str(operand: InferenceResult | None) -> bool:
         """Return True if the operand in argument is a literal string."""
         return isinstance(operand, nodes.Const) and operand.name == "str"
+    
+    @staticmethod
+    def _is_node_explicit_str_concatenation(node: nodes.NodeNG) -> bool:
+        """Return True if the node represents an explicitly concatenated string."""
+        if not isinstance(node, nodes.BinOp):
+            return False
+        return (
+            LoggingChecker._is_operand_literal_str(node.left)
+            or LoggingChecker._is_node_explicit_str_concatenation(node.left)
+        ) and (
+            LoggingChecker._is_operand_literal_str(node.right)
+            or LoggingChecker._is_node_explicit_str_concatenation(node.right)
+        )
 
     def _check_call_func(self, node: nodes.Call) -> None:
         """Checks that function call is not format_string.format()."""

--- a/tests/functional/l/logging/logging_not_lazy.py
+++ b/tests/functional/l/logging/logging_not_lazy.py
@@ -9,10 +9,10 @@ var = "123"
 var_name = "Var:"
 # Statements that should be flagged:
 renamed_logging.warn("%s, %s" % (4, 5))  # [logging-not-lazy]
+renamed_logging.warn("Var: " + var)  # [logging-not-lazy]
 renamed_logging.exception("%s" % "Exceptional!")  # [logging-not-lazy]
 renamed_logging.log(renamed_logging.INFO, "msg: %s" % "Run!")  # [logging-not-lazy]
 renamed_logging.log(renamed_logging.INFO, "Var: " + var)  # [logging-not-lazy]
-renamed_logging.warn("%s" + " the rest of a single string")  # [logging-not-lazy]
 renamed_logging.log(renamed_logging.INFO, var_name + var)  # [logging-not-lazy]
 
 # Statements that should not be flagged:
@@ -21,6 +21,11 @@ renamed_logging.log(renamed_logging.INFO, "msg: %s", "Run!")
 logging.warn("%s, %s" % (4, 5))
 logging.log(logging.INFO, "msg: %s" % "Run!")
 logging.log("Var: " + var)
+# Explicit string concatenations are fine:
+renamed_logging.warn("%s" + " the rest of a single string")
+renamed_logging.warn("Msg: " + "%s", "first piece " + "second piece")
+renamed_logging.warn("first" + "second" + "third %s", "parameter")
+renamed_logging.warn(("first" + "second" + "third %s"))
 
 # Regression crash test for incorrect format call
 renamed_logging.error(

--- a/tests/functional/l/logging/logging_not_lazy.txt
+++ b/tests/functional/l/logging/logging_not_lazy.txt
@@ -1,8 +1,8 @@
 logging-not-lazy:11:0:11:39::Use lazy % formatting in logging functions:UNDEFINED
-logging-not-lazy:12:0:12:48::Use lazy % formatting in logging functions:UNDEFINED
-logging-not-lazy:13:0:13:61::Use lazy % formatting in logging functions:UNDEFINED
-logging-not-lazy:14:0:14:56::Use lazy % formatting in logging functions:UNDEFINED
-logging-not-lazy:15:0:15:59::Use lazy % formatting in logging functions:UNDEFINED
+logging-not-lazy:12:0:12:35::Use lazy % formatting in logging functions:UNDEFINED
+logging-not-lazy:13:0:13:48::Use lazy % formatting in logging functions:UNDEFINED
+logging-not-lazy:14:0:14:61::Use lazy % formatting in logging functions:UNDEFINED
+logging-not-lazy:15:0:15:56::Use lazy % formatting in logging functions:UNDEFINED
 logging-not-lazy:16:0:16:57::Use lazy % formatting in logging functions:UNDEFINED
-bad-format-string:27:4:27:27::Invalid format string:UNDEFINED
-logging-format-interpolation:27:4:27:27::Use lazy % formatting in logging functions:UNDEFINED
+bad-format-string:32:4:32:27::Invalid format string:UNDEFINED
+logging-format-interpolation:32:4:32:27::Use lazy % formatting in logging functions:UNDEFINED

